src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -135,51 +134,63 @@ private IEnumerable<MethodBase> FunctionLookup(IList args)
-                        m => m.Name == methodName && m.GetParameters().Length == args.Count).ToList();
-
-                    // Check for instance methods
-                    if (mi == null || !mi.Any())
+                    var modules = ProtoFFI.DLLFFIHandler.Modules.Values.OfType<ProtoFFI.CLRDLLModule>();
+                    var assemblies = modules.Select(m => m.Assembly ?? (m.Module?.Assembly)).Where(m => m != null);
+                    foreach (var asm in assemblies)
                     {
-                        mi = type.GetMethods(BindingFlags.Public | BindingFlags.Instance).Where(
-                            m => m.Name == methodName && m.GetParameters().Length + 1 == args.Count).ToList();
-                    }
+                        var type = asm.GetType(className);
+                        if (type == null) continue;
 
-                    // Check for property getters
-                    if (mi == null || !mi.Any())
-                    {
-                        var prop = type.GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance).Where(
-                                p => p.Name == methodName).FirstOrDefault();
+                        // There should be a way to get the exact method after matching parameter types for a node
+                        // using its function descriptor. AST isn't sufficient for parameter type info.
+                        // Fist check for static methods
+                        mi = type.GetMethods(BindingFlags.Public | BindingFlags.Static).Where(
+                            m => m.Name == methodName && m.GetParameters().Length == args.Count).ToList();
 
-                        if (prop != null)
+                        // Check for instance methods
+                        if (mi == null || !mi.Any())
                         {
-                            mi = prop.GetAccessors().ToList();
+                            mi = type.GetMethods(BindingFlags.Public | BindingFlags.Instance).Where(
+                                m => m.Name == methodName && m.GetParameters().Length + 1 == args.Count).ToList();
                         }
-                    }
 
-                    // TODO: Add check for constructorinfo objects
+                        // Check for property getters
+                        if (mi == null || !mi.Any())
+                        {
+                            var prop = type
+                                .GetProperties(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance).Where(
+                                    p => p.Name == methodName).FirstOrDefault();
+
+                            if (prop != null)
+                            {
+                                mi = prop.GetAccessors().ToList();
+                            }
+                        }
+
+                        // TODO: Add check for constructorinfo objects
+
+                        if (mi != null)
```

Comment:
I found that sometimes the lookups above can return an empty list and then we end up with an empty list in the method cache.